File: listuser_fe/services/userService.ts
/*
 * Copyright(C) 2026 Luvina Software Company
 *
 * userService.ts, 11/08/2026 DoanQuangMinh
 */

import { User } from "@/types/user";

const API_URL = "http://localhost:8080/api/users";

/**
 * Lấy danh sách tất cả người dùng từ User API.
 *
 * @return Promise chứa danh sách người dùng.
 * @throws Error khi API trả về response không thành công.
 * @author DoanQuangMinh
 */
export async function getUsers(): Promise<User[]> {
  const response = await fetch(API_URL, {
    cache: "no-store",
  });

  if (!response.ok) {
    throw new Error("Failed to fetch users.");
  }

  return response.json();
}

/**
 * Lấy thông tin chi tiết của người dùng theo ID.
 *
 * @param id ID của người dùng cần lấy thông tin.
 * @return Promise chứa thông tin người dùng.
 * @throws Error khi API trả về response không thành công.
 * @author DoanQuangMinh
 */
export async function getUserById(id: string): Promise<User> {
  const response = await fetch(`${API_URL}/${id}`, {
    cache: "no-store",
  });

  if (!response.ok) {
    throw new Error("Failed to fetch user.");
  }

  return response.json();
}

File: listuser_be/src/main/java/listuser_be/service/UserService.java
/*
 * Copyright(C) 2026 Luvina Software Company
 *
 * UserService.java, 11/08/2026 DoanQuangMinh
 */

package listuser_be.service;

import java.util.List;

import listuser_be.entity.User;

/**
 * Interface định nghĩa các nghiệp vụ xử lý User.
 *
 * @author DoanQuangMinh
 */
public interface UserService {

    /**
     * Lấy danh sách tất cả User.
     *
     * @return danh sách User
     */
    List<User> getAllUsers();

    /**
     * Lấy User theo ID.
     *
     * @param id mã User cần tìm
     * @return User tương ứng với ID
     */
    User getUserById(Integer id);
}

